mono/mini/mini-windows.c

```diff
@@ -152,6 +152,42 @@ get_win32_restore_stack (void)
 
 	return start;
 }
+#elif HAVE_API_SUPPORT_WIN32_RESET_STKOFLW && TARGET_ARM64
+static gpointer
+get_win32_restore_stack(void)
+{
+	static guint8* start = NULL;
+	guint8* code;
+
+	if (start)
+		return start;
+
+	const int size = 64;
+
+	/* restore_stack (void) */
+	start = code = mono_global_codeman_reserve(size);
+
+	arm_stpx_pre(code, ARMREG_FP, ARMREG_LR, ARMREG_SP, -0x10);
+	arm_movx(code, ARMREG_FP, ARMREG_SP);
+	arm_adrpx(code, ARMREG_R8, _resetstkoflw);
+	arm_addx(code, ARMREG_R8, ARMREG_R8, ((int)_resetstkoflw) & 0xFFF);
+	arm_blrx(code, ARMREG_R8);
+	arm_adrpx(code, ARMREG_R8, mono_tls_get_jit_tls_extern);
+	arm_addx(code, ARMREG_R8, ARMREG_R8, ((int)mono_tls_get_jit_tls_extern) & 0xFFF);
```

Comment:
It might be worth adding `arm_adrl` macro to `arm64-codegen.h` to prevent repeating yourself and preventing accidental typos. @joncham, do you think that's ok?